File: lib/widgets/alarm_tile.dart
import 'package:flutter/material.dart';

class AlarmTile extends StatelessWidget {
  final String time;
  final List<String> days;
  final bool enabled;
  final ValueChanged<bool>? onToggle;

  const AlarmTile({
    super.key,
    required this.time,
    required this.days,
    required this.enabled,
    this.onToggle,
  });

  @override
  Widget build(BuildContext context) {
    return Container(
      margin: const EdgeInsets.symmetric(vertical: 8),
      padding: const EdgeInsets.symmetric(horizontal: 20, vertical: 16),
      decoration: BoxDecoration(
        color: Colors.grey[900],
        borderRadius: BorderRadius.circular(20),
      ),
      child: Row(
        children: [
          // 시간 + 요일 정보
          Expanded(
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                Text(
                  time,
                  style: TextStyle(
                    color: enabled ? Colors.white : Colors.grey,
                    fontSize: 28,
                    fontWeight: FontWeight.bold,
                  ),
                ),
                const SizedBox(height: 4),
                Text(
                  days.join(' '),
                  style: TextStyle(
                    color: enabled ? Colors.white70 : Colors.grey,
                    fontSize: 14,
                  ),
                ),
              ],
            ),
          ),

          // 스위치
          Switch(
            value: enabled,
            onChanged: onToggle,
            activeColor: Colors.grey,
          ),
        ],
      ),
    );
  }
}

File: app/src/main/java/lab/p4c/nextup/ui/widget/AlarmTile.kt
package lab.p4c.nextup.ui.widget

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Surface
import androidx.compose.material3.Switch
import androidx.compose.material3.SwitchDefaults
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp

@Composable
fun AlarmTile(
    time: String,
    days: List<String>,
    enabled: Boolean,
    onToggle: ((Boolean) -> Unit)?
) {
    val bg = Color(0xFF212121) // Flutter Colors.grey[900]
    val timeColor = if (enabled) Color.White else Color(0xFF9E9E9E)
    val daysColor = if (enabled) Color(0xB3FFFFFF) /* white70 */ else Color(0xFF9E9E9E)

    Surface(
        color = bg,
        shape = RoundedCornerShape(20.dp),
        modifier = Modifier
            .fillMaxWidth()
            .padding(vertical = 8.dp)
    ) {
        Row(
            modifier = Modifier
                .padding(horizontal = 20.dp, vertical = 16.dp),
            horizontalArrangement = Arrangement.SpaceBetween
        ) {
            Column(Modifier.weight(1f)) {
                Text(
                    text = time,
                    color = timeColor,
                    fontSize = androidx.compose.material3.MaterialTheme.typography.headlineMedium.fontSize, // ~28sp
                    fontWeight = FontWeight.Bold
                )
                Spacer(Modifier.height(4.dp))
                Text(
                    text = days.joinToString(" "),
                    color = daysColor,
                    fontSize = androidx.compose.material3.MaterialTheme.typography.bodySmall.fontSize
                )
            }

            Switch(
                checked = enabled,
                onCheckedChange = onToggle,
                colors = SwitchDefaults.colors(
                    checkedThumbColor = Color(0xFF9E9E9E), // Flutter activeColor: Colors.grey
                    checkedTrackColor = Color(0xFFBDBDBD)
                )
            )
        }
    }
}
